osu.Game.Rulesets.Taiko/UI/KiaiHitExplosion.cs

```diff
@@ -1,71 +1,53 @@
 // See the LICENCE file in the repository root for full licence text.
 
-using osuTK;
 using osu.Framework.Allocation;
 using osu.Framework.Graphics;
 using osu.Framework.Graphics.Containers;
-using osu.Framework.Graphics.Effects;
-using osu.Framework.Graphics.Shapes;
-using osu.Game.Graphics;
 using osu.Game.Rulesets.Objects.Drawables;
 using osu.Game.Rulesets.Taiko.Objects;
+using osu.Game.Skinning;
+using osuTK;
 
 namespace osu.Game.Rulesets.Taiko.UI
 {
-    public class KiaiHitExplosion : CircularContainer
+    public class KiaiHitExplosion : Container
     {
         public override bool RemoveWhenNotAlive => true;
 
+        [Cached(typeof(DrawableHitObject))]
         public readonly DrawableHitObject JudgedObject;
-        private readonly HitType type;
 
-        public KiaiHitExplosion(DrawableHitObject judgedObject, HitType type)
+        private readonly HitType hitType;
+
+        private SkinnableDrawable skinnable;
+
+        public override double LifetimeStart => skinnable.Drawable.LifetimeStart;
+
+        public override double LifetimeEnd => skinnable.Drawable.LifetimeEnd;
+
+        public KiaiHitExplosion(DrawableHitObject judgedObject, HitType hitType)
         {
             JudgedObject = judgedObject;
-            this.type = type;
+            this.hitType = hitType;
 
             Anchor = Anchor.CentreLeft;
             Origin = Anchor.Centre;
 
             RelativeSizeAxes = Axes.Both;
             Size = new Vector2(TaikoHitObject.DEFAULT_SIZE, 1);
-
-            Blending = BlendingParameters.Additive;
-
-            Masking = true;
-            Alpha = 0.25f;
-
-            Children = new[]
-            {
-                new Box
-                {
-                    RelativeSizeAxes = Axes.Both,
-                    Alpha = 0,
-                    AlwaysPresent = true
-                }
-            };
         }
 
         [BackgroundDependencyLoader]
-        private void load(OsuColour colours)
+        private void load()
         {
-            EdgeEffect = new EdgeEffectParameters
-            {
-                Type = EdgeEffectType.Glow,
-                Colour = type == HitType.Rim ? colours.BlueDarker : colours.PinkDarker,
-                Radius = 60,
-            };
+            Child = skinnable = new SkinnableDrawable(new TaikoSkinComponent(TaikoSkinComponents.TaikoExplosionKiai), _ => new DefaultKiaiHitExplosion(hitType));
         }
 
-        protected override void LoadComplete()
-        {
-            base.LoadComplete();
-
-            this.ScaleTo(new Vector2(1, 3f), 500, Easing.OutQuint);
-            this.FadeOut(250);
-
-            Expire(true);
-        }
+        /// <summary>
+        /// Helper function to use when an explosion is not desired.
+        /// Lifetime is set to avoid accumulating empty drawables in the parent container.
+        /// </summary>
+        public static Drawable EmptyExplosion() => Empty().With(d => d.LifetimeEnd = double.MinValue);
```

Comment:
maybe just inline this at the point of usage? don't think there's much value in leaving it here unless it's potentially going to be reused somewhere?

also might look better as `new Drawable { Lifetime End = ...`

Reply:
That's how I had it at the start but wasn't sure if it didn't look overly hideous. Will inline locally instead.